src/main/java/org/killbill/billing/plugin/adyen/client/AdyenConfigProperties.java

```diff
@@ -90,22 +94,29 @@
 
     private final Period pending3DsPaymentExpirationPeriod;
 
-    public AdyenConfigProperties(final Properties properties) {
+    private final String currentRegion;
+
+    public AdyenConfigProperties(final Properties properties, final Properties globalProperties) {
+        this.currentRegion = PluginEnvironmentConfig.getRegion(globalProperties);
```

Comment:
An alternative implementation would be looking for properties with the region as the prefix, i.e. `snc1.org.killbill.billing.plugin.adyen.paymentUrl`, if that doesn't exists, or the region is undefined, the plugin fallbacks to `org.killbill.billing.plugin.adyen.paymentUrl`.

The main benefit being we wouldn't need to work with the cumbersome map like syntax.

Also, I'd add specific parameter for the region, instead of extracting it from a globalProperties. That way we shield the class from changes on how the region is looked up. It also makes the constructor more explicit.

I think it would also make sense to keep an overloaded constructor with a single properties param, where the region would always be `null`. Would make instantiating this class in tests easier.